bitcoin/src/crypto/sighash.rs

```diff
@@ -76,8 +76,12 @@ impl SegwitV0Sighash {
     fn from_engine(e: sha256d::HashEngine) -> Self { Self(sha256d::Hash::from_engine(e)) }
 }
 
-sha256t_tag! {
-    pub struct TapSighashTag = hash_str("TapSighash");
+/// Tag for the `TapSighash`.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
Oh this is SSOT violation. But we can avoid it if we drop these entirely and turn derives on `sha256t::Hash` into manual impls with the correct bounds.